include/alpaka/dev/DevSycl.hpp

```diff
@@ -0,0 +1,220 @@
+            ALPAKA_FN_HOST auto operator==(DevSycl const & rhs) const -> bool = default;
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST auto operator!=(DevSycl const & rhs) const -> bool = default;
+            //-----------------------------------------------------------------------------
+            ~DevSycl() = default;
+
+        public:
+            cl::sycl::device m_Device;
+            cl::sycl::queue m_Queue;
+        };
+    }
+
+    namespace dev
+    {
+        namespace traits
+        {
+            //#############################################################################
+            //! The SYCL device name get trait specialization.
+            template<>
+            struct GetName<
+                dev::DevSycl>
+            {
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST static auto getName(
+                    dev::DevSycl const & dev)
+                -> std::string
+                {
+                    // get_info returns std::string in this case
+                    return dev.m_Device.get_info<
+                        cl::sycl::info::device::name>();
+                }
+            };
+
+            //#############################################################################
+            //! The SYCL device available memory get trait specialization.
+            template<>
+            struct GetMemBytes<
+                dev::DevSycl>
+            {
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST static auto getMemBytes(
+                    dev::DevSycl const & dev)
+                -> std::size_t
+                {
+                    // get_info returns cl_ulong in this case
+                    return dev.m_Device.get_info<
+                        cl::sycl::info::device::global_mem_size>();
+                }
+            };
+
+            //#############################################################################
+            //! The SYCL device free memory get trait specialization. Note that
+            //! this function will always return 0 as there is no way in SYCL
+            //! or OpenCL to query free memory.
+            template<>
+            struct GetFreeMemBytes<
+                dev::DevSycl>
+            {
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST static auto getFreeMemBytes(
+                    dev::DevSycl const & dev)
+                -> std::size_t
+                {
+                    // FIXME: There is no way in either SYCL or OpenCL to
+                    // query free memory.
+                    throw std::runtime_error{"Querying free device memory not supported on SYCL platforms"};
```

Comment:
This somehow conflicts with the documentation. I am not yet sure if I would prefer the pessimization of returning `0u´ over the exception.

Reply:
Fixed the documentation. I'm leaving it as an exception for now unless you prefer the `0u`?